ingester/src/querier_handler.rs

```diff
@@ -0,0 +1,196 @@
+        let stream = query(&exc, batch, Predicate::default(), Selection::All)
+            .await
+            .unwrap();
+        let output_batches = datafusion::physical_plan::common::collect(stream)
+            .await
+            .unwrap();
+
+        // verify data: all rows and columns should be returned
+        let expected = vec![
+            "+-----------+------+-----------------------------+",
+            "| field_int | tag1 | time                        |",
+            "+-----------+------+-----------------------------+",
+            "| 70        | UT   | 1970-01-01T00:00:00.000020Z |",
+            "| 10        | VT   | 1970-01-01T00:00:00.000010Z |",
+            "| 1000      | WA   | 1970-01-01T00:00:00.000008Z |",
+            "+-----------+------+-----------------------------+",
+        ];
+        assert_batches_eq!(&expected, &output_batches);
+    }
+
+    #[tokio::test]
+    async fn test_query_with_filter() {
+        test_helpers::maybe_start_logging();
+
+        // create input data
+        let batches = create_one_record_batch_with_influxtype_no_duplicates().await;
+        //let tombstones = vec![create_tombstone(1, 1, 1, 1, 0, 200000, "tag1=UT")];
+
+        // build queryable batch from the input batches
+        let batch = make_queryable_batch("test_table", 1, batches);
+
+        // make filters
+        // Only read 2 columns: "tag1" and "time"
+        let selection = Selection::Some(&["tag1", "time"]);
+
+        // tag1=VT
+        let expr = col("tag1").eq(lit("VT"));
+        let pred = PredicateBuilder::default().add_expr(expr).build();
+
+        let exc = Executor::new(1);
+        let stream = query(&exc, batch, pred, selection).await.unwrap();
+        let output_batches = datafusion::physical_plan::common::collect(stream)
+            .await
+            .unwrap();
+
+        // verify data: 2  columns and one row of "tag1=VT" should be returned
+        let expected = vec![
+            "+------+-----------------------------+",
+            "| tag1 | time                        |",
+            "+------+-----------------------------+",
+            "| VT   | 1970-01-01T00:00:00.000010Z |",
+            "+------+-----------------------------+",
+        ];
+        assert_batches_eq!(&expected, &output_batches);
+    }
+
+    #[tokio::test]
+    async fn test_query_with_filter_with_delete() {
+        test_helpers::maybe_start_logging();
+
+        // create input data
+        let batches = create_one_record_batch_with_influxtype_no_duplicates().await;
+        let tombstones = vec![create_tombstone(1, 1, 1, 1, 0, 200000, "tag1=UT")];
+
+        // build queryable batch from the input batches
+        let batch = make_queryable_batch_with_deletes("test_table", 1, batches, tombstones);
+
+        // make filters
+        // Only read 2 columns: "tag1" and "time"
+        let selection = Selection::Some(&["tag1", "time"]);
+
+        // tag1=VT
+        let expr = col("tag1").eq(lit("UT"));
```

Comment:
I think this filter says "UT" rather than "VT"

```suggestion
        let expr = col("tag1").eq(lit("VT"));
```

Reply:
I will modify comment to make it `tag`=UT` to have the result as expected